src/core/primitives.cc

```diff
@@ -404,13 +405,17 @@ std::string PolyhedronNode::toString() const
 
 std::unique_ptr<const Geometry> PolyhedronNode::createGeometry() const
 {
-  auto p = std::make_unique<PolySet>(3);
-  p->setConvexity(this->convexity);
-  p->vertices=this->points;
-  p->indices=this->faces;
-  for (auto &poly : p->indices)
+  PolySetBuilder builder(this->points.size(), this->faces.size());
```

Comment:
going through `PolySetBuilder` will lose manifoldness.

Reply:
Riight, this is the wrong hammer for this job; we need something that cleans up a user-specified polyhedron. I'll think about it, perhaps just tessellate to triangles early for user polyhedrons.